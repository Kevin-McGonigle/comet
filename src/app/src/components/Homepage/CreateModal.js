import React, { useState } from 'react';
import styles from './CreateModal.css';
import classnames from 'classnames';
import { Textarea, Alert, Dialog, Pane, Tablist, Tab, IconButton, TextInput } from 'evergreen-ui';
import upload_files from '../../api/API';

const cx = args => classnames(styles, args);

const createFileInformation = (fileNames, fileContent) => {
    const fileData = fileNames.map((file, ind) => {
        return new File([fileContent[ind]], file, { type: "text/plain", lastModified: "test" })
    });
    return fileData;
}

export const shapeFileData = (fileData) => {
    const shapedData = fileData.map((file) => {
        const data = readFile(file);
        return {
            name: file.name,
            lastModified: file.lastModified,
            size: file.size,
            type: file.type,
            content: data,
        };
    });
    return shapedData;
}

const CreateModal = props => {
    const {
        alertInfo,
        createModal,
        createModalOnConfirmHandler,
        createModalOnCloseHandler,
        createModalOnFailureHandler,
        setInheritanceTree,
        setAlertSuccess,
        setAlertDanger,
        setFileData,
        history,
    } = props; 

    const [fileTabs, setFileTabs] = useState({
        selectedIndex: 0,
        tabs: ['File'],
        tabContent: [''],
    })

    const [newFileDialog, setNewFileDialog] = useState({
        isShown: false,
        fileName: '',
    })

    const onTextAreaChangeHandler = (event) => {
        const newFileContent = fileTabs.tabContent;
        newFileContent[fileTabs.selectedIndex] = event.target.value;
        setFileTabs({ ...fileTabs, tabContent: newFileContent });
    }

    const onPlusIconHandler = () => {
        setNewFileDialog({ ...newFileDialog, isShown: true });
    }

    const onDeleteIconHandler = () => {
        const selectedInd = fileTabs.selectedIndex;
        const newTabs = fileTabs.tabs.splice(selectedInd, 1);
        const newContent = fileTabs.tabContent.splice(selectedInd, 1);

        setFileTabs({
            selectedIndex: 0,
            tabs: newTabs,
            tabContent: newContent,
        });
    }

    const onEditIconHandler = () => {

    }

    const upload = () => {
        createModalOnConfirmHandler();
        // need to shape data into file 
        const fileData = createFileInformation(fileTabs.tabs, fileTabs.tabContent)
        const shapedData = shapeFileData(fileData);
        upload_files(fileData).then((data) => {
            if (data) {
                setFileData(shapedData);
                setInheritanceTree(data);
                setAlertSuccess("Uploaded succesfully!")
                history.push('/metrics');
            } else {
                createModalOnFailureHandler();
                setAlertDanger("Could not upload: " + data.statusText)
            }
        });
    }

    const onDialogCloseHandler = () => {
        const newTabs = fileTabs.tabs;
        newTabs.push(newFileDialog.fileName)
        const newTabContent = fileTabs.tabContent;
        newTabContent.push('');

        setFileTabs({
            ...fileTabs,
            tabContent: newTabContent,
            tabs: newTabs
        });
        setNewFileDialog({ isShown: false, fileName: '' });
    }

    const onTextInputChange = (event) => {
        const value = event.target.value;
        setNewFileDialog({ ...newFileDialog, fileName: value });
    }

    const createAlert = () => {
        if (alertInfo.show){
            return (
                <Alert
                    id={`${alertInfo.intent}Alert`}
                    intent={alertInfo.intent}
                    hasIcon
                    title={alertInfo.title}
                    marginTop={2}
                    marginBottom={8}
                />
            )}
    }

    return (
        <Dialog
            isShown={createModal.isOpen}
            title="Create"
            onConfirm={upload}
            isConfirmLoading={createModal.isLoading}
            confirmLabel={createModal.isLoading ? "Creating & Uploading" : "Upload"}
            onCloseComplete={createModalOnCloseHandler}
        >
            { createAlert() }

            <div className={cx('fileTab')}>
                <div className={cx('icons')}>
                    <Dialog
                        isShown={newFileDialog.isShown}
                        title="Create new file"
                        onCloseComplete={onDialogCloseHandler}
                        confirmLabel="Create"
                    >
                        <TextInput
                            name="File name"
                            placeholder="Filename.py"
                            onChange={onTextInputChange}
                        />
                    </Dialog>

                    <IconButton icon="plus" onClick={onPlusIconHandler}/>
                    <IconButton icon="edit" onClick={onEditIconHandler}/>
                    <IconButton icon="trash" intent="danger" onClick={onDeleteIconHandler}/>
                </div>
            
                <Pane height={16}>
                    <Tablist marginBottom={14} flexBasis={240} marginRight={24}>
                        { fileTabs.tabs.map((tab, index) => (
                            <Tab
                                key={tab}
                                id={tab}
                                onSelect={() => setFileTabs({ ...fileTabs, selectedIndex: index })}
                                isSelected={index === fileTabs.selectedIndex}
                                aria-controls={`panel-${tab}`}
                            >
                                {tab}
                            </Tab>
                        ))}
                    </Tablist>
                </Pane>
            </div>

            <div className={cx('createModeContainer')}>
                <div className={cx('textAreaContainer')}>
                    <Textarea 
                        name="textArea"
                        id="textAreaId"
                        placholder="Input your code here.."
                        onChange={onTextAreaChangeHandler}
                        value={fileTabs.tabContent[fileTabs.selectedIndex]}
                    />
                </div>
            </div>
        </Dialog>
    );
};

export default CreateModal;




