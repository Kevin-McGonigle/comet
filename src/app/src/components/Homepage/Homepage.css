html { 
    height: 100%;
    margin:0; 
    padding:0; 
}

body {
    min-height: 100%;
}

.homepageContainer {
    height:100vh;
    width: 100%;
    text-align: center;
    position: relative;
    overflow: hidden;
    background-color:#11182f ;
}

.topBar {
    text-align: right;
    margin-right: 20px;
    color: #ffff;
}

.logoContainer {
    margin: 15%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.logo {
    transform: translate(0, -75%);
}

h1 { 
    z-index: 5;
    color: #ffffff; 
    font-family: 'Open Sans', serif; 
    font-size: 65px; 
    font-weight: normal; 
    line-height: 48px; 
    margin: 0;
}

h2 { 
    z-index: 5;
    color: #ffffff;
    font-family: 'Open Sans', sans-serif;
    font-size: 30px;
    font-weight: 300;
    line-height: 32px;
}
