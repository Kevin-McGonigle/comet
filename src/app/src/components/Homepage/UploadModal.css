.uploadModalContainer {
    display: flex;
    flex-direction: column;
}

.filePicker {
    justify-content: center;
}

.fileItemContainer {
    display: flex;
    align-items: center;
    background-color: #D4EEE2;
    margin: 8px 0px 4px 0px;
}
.fileItems {
    flex-grow: 1;
    margin: 1em 3em;
	color: #131516;
	border-left: 2px solid #131516;
    padding-left: 1em; 
}

.deleteIcon {
    float: right;
    padding-right: 1em; 
    padding-left: 1em; 
    cursor: pointer;
}