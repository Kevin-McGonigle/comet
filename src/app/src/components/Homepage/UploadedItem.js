import React from 'react';
import styles from './UploadModal.css';
import classnames from 'classnames';
import { Icon } from 'evergreen-ui';

const cx = args => classnames(styles, args)

const UploadedItem = (props) => {
    const {
        name,
        fileType,
        size,
        deleteOnClickHandler,
    } = props;

    return (
        <div className={cx('fileItemContainer')}>
            <div className={cx('fileItems')}>
                <div className={'fileItem'}>Name: <i>{ name }</i></div>
                <div className={'fileItem'}>Size: <i>{ size }</i></div>
                <div className={'fileItem'}>File Type: <i>{ fileType }</i></div>
            </div>
            <div className={cx('deleteIcon')}>
                <Icon icon="trash" id='deleteIcon' onClick={name => deleteOnClickHandler(name)} />
            </div>
        </div>
    )
}

export default UploadedItem;