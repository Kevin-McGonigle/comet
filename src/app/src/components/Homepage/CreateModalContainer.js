import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import CreateModal from './CreateModal';
import { actions as alertActions } from '../../store/alert/alert';

const mapStateToProps = state => ({
    alertInfo: state.alert,
})

const mapDispatchToProps = dispatch => bindActionCreators(alertActions, dispatch)

export const CreateModalContainer = connect(mapStateToProps, mapDispatchToProps)(CreateModal)



