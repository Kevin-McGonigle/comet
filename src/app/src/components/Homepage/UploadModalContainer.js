import {connect} from 'react-redux';
import UploadModal from './UploadModal';
import {actions as alertActions} from '../../store/alert/alert';
import {actions as metricsActions} from '../../store/metrics/metrics';
import {actions as fileDataActions} from '../../store/fileData/fileData';

const mapStateToProps = state => ({
    alertInfo: state.alert,
    fileData: state.fileData.files,
})

const mapDispatchToProps = (dispatch) => {
    return {
        setMetrics: (metrics) => dispatch(metricsActions.setMetrics(metrics)),
        setFileData: (data) => dispatch(fileDataActions.setFileData(data)),
        setAlertSuccess: (text) => dispatch(alertActions.setAlertSuccess(text)),
        setAlertDanger: (text) => dispatch(alertActions.setAlertDanger(text)),
        setAlertNone: () => dispatch(alertActions.setAlertNone()),
    }
};

export const UploadModalContainer = connect(mapStateToProps, mapDispatchToProps)(UploadModal)



