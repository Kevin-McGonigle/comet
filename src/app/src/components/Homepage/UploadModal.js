import React from 'react';
import styles from './UploadModal.css';
import classnames from 'classnames';
import {Alert, Dialog, FilePicker} from 'evergreen-ui';
import UploadedItem from './UploadedItem';
import upload_files from '../../api/API';

const cx = args => classnames(styles, args)

export const removeFileFromUploadedFiles = (files, name) => {
    return Object.values(files).reduce((acc, file) => {
        if (file.name !== name) {
            acc.push(file);
        }
        return acc;
    }, []);
}

export async function readFile(file) {
    const fileReader = new FileReader();
    const promise = new Promise((resolve, reject) => {
        fileReader.onerror = () => {
            fileReader.abort();
            reject('Problem parsing input files');
        };
        fileReader.onload = () => {
            resolve(fileReader.result);
        };
        fileReader.readAsText(file);
    });
    return await promise;
}

export const shapeFileData = (fileData) => {
    const shapedData = fileData.map((file) => {
        const data = readFile(file);
        return {
            name: file.name,
            lastModified: file.lastModified,
            size: file.size,
            type: file.type,
            content: data,
        };
    });
    return shapedData;
}

const UploadModal = props => {
    const {
        history,
        fileData,
        alertInfo,
        uploadModal,
        uploadModalOnConfirmHandler,
        uploadModalOnCloseHandler,
        setAlertSuccess,
        setAlertDanger,
        setFileData,
        setInheritanceTree,
    } = props;

    const fileItemDeleteOnClickHandler = (name) => {
        const updatedUploadedFiles = removeFileFromUploadedFiles(fileData, name);
        setFileData(updatedUploadedFiles);
    }

    const createFileItem = (file) => {
        return (
            <UploadedItem
                name={file.name}
                size={file.size}
                fileType={file.type}
                deleteOnClickHandler={() => fileItemDeleteOnClickHandler(file.name)}
            />
        )
    }

    const createAlert = () => {
        if (alertInfo.show) {
            return (
                <Alert
                    id={`${alertInfo.intent}Alert`}
                    intent={alertInfo.intent}
                    title={alertInfo.title}
                    hasIcon
                    marginTop={2}
                    marginBottom={8}
                />
            );
        }
    }

    const onChangeHandler = (files) => {
        setFileData(files);
        setAlertSuccess('Successfully added');
    };

    const upload = () => {
        uploadModalOnConfirmHandler();
        upload_files(fileData).then((data) => {
            if (data) {
                const shapedData = shapeFileData(fileData);
                setFileData(shapedData);
                setInheritanceTree(data);
                setAlertSuccess("Uploaded succesfully!")
                history.push('/metrics');
            } else {
                setAlertDanger("Could not upload, please try again!")
            }
        });
    }

    return (
        <Dialog
            isShown={uploadModal.isOpen}
            id="uploadModal"
            title="Upload"
            isConfirmLoading={uploadModal.isLoading}
            confirmLabel={uploadModal.isLoading ? "Uploading.." : 'Upload'}
            onConfirm={upload}
            onCloseComplete={uploadModalOnCloseHandler}
        >
            <div className={cx('uploadModalContainer')}>
                <div className={cx('filePicker')}>
                    {createAlert()}
                    <FilePicker
                        id='filePicker'
                        multiple
                        width={'100%'}
                        height={24}
                        onChange={onChangeHandler}
                        placeholder={"Choose file or folder"}
                    />
                </div>

                {fileData && (
                    <div className={cx('fileListContainer')}>
                        {Object.values(fileData).map(file => createFileItem(file))}
                    </div>
                )}
            </div>
        </Dialog>
    );
};

export default UploadModal;