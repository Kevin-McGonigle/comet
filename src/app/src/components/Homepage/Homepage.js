import React, {useState} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import styles from './Homepage.css';
import classnames from 'classnames';
import comet from './comet.png';
import {Button} from 'evergreen-ui';
import {UploadModalContainer} from './UploadModalContainer';
import {CreateModalContainer} from './CreateModalContainer';
import {actions as alertActions} from '../../store/alert/alert';

const cx = args => classnames(styles, args)

const Homepage = props => {
    const {
        setAlertNone,
    } = props;
    const [createModal, setCreateModal] = useState({ isOpen: false, isLoading: false});
    const [uploadModal, setUploadModal] = useState({ isOpen: false, isLoading: false });

    // Handlers stop the creation of new functions each render 
    const createButtonOnClickHandler = () => setCreateModal({isOpen: true, isLoading: false});
    const createModalOnCloseHandler = () => {
        setAlertNone();
        setCreateModal({isOpen: false, isLoading: false});
    }
    const createModalOnConfirmHandler = () => {
        setAlertNone();
        setCreateModal({isOpen: true, isLoading: true})
    }

    const createModalOnFailureHandler = () => {
        setCreateModal({isOpen: true, isLoading: false})
    }

    const uploadButtonOnClickHandler = () => {
        setUploadModal({isOpen: true, isLoading: false});
    }

    const uploadModalOnCloseHandler = () => {
        setAlertNone();
        setUploadModal({isOpen: false, isLoading: false});
    }

    const uploadModalOnConfirmHandler = () => {
        setAlertNone();
        setUploadModal({isOpen: true, isLoading: true})
    }


    return (
        <div className={cx('homepageContainer')}>
            <div className={cx('topBar')}>
                <p>by James Miles & Kevin McGonigle</p>
            </div>
            <div className={cx('logoContainer')}>
                <h1>comet.</h1>
                <h2>code metrics - made simple</h2>
                <div className={cx('logo')}>
                    <img src={comet} width="200px" height="200px" alt={"Comet logo."}/>

                    <div>
                        <CreateModalContainer
                            createModal={createModal}
                            createModalOnConfirmHandler={createModalOnConfirmHandler}
                            createModalOnCloseHandler={createModalOnCloseHandler}
                            createModalOnFailureHandler={createModalOnFailureHandler}
                        />

                        <UploadModalContainer
                            uploadModal={uploadModal}
                            uploadModalOnConfirmHandler={uploadModalOnConfirmHandler}
                            uploadModalOnCloseHandler={uploadModalOnCloseHandler}
                        />

                        <Button
                            id='createButton'
                            className={cx('createButton')}
                            onClick={createButtonOnClickHandler}
                            marginRight={12}
                            height={48}>
                            Create
                        </Button>

                        <Button
                            id='uploadButton'
                            className={cx('uploadButton')}
                            onClick={uploadButtonOnClickHandler}
                            marginRight={12}
                            height={48}>
                            Upload
                        </Button>
                    </div>
                </div>
            </div>
        </div>
    );
};

const mapDispatchToProps = dispatch => bindActionCreators(alertActions, dispatch);

export default connect(null, mapDispatchToProps)(Homepage);