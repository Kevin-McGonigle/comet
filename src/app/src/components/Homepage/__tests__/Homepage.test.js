import React from 'react';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store' 
import Homepage from '../Homepage';
import { shallow, mount } from 'enzyme';
import "../../../setupTests"

jest.mock('react-router-dom', () => ({
    useHistory: () => ({
      push: jest.fn(),
    }),
}));

const initialState = {
    alert: {
        id: 'something',
        title: 'something',
        show: 'something',
        intent: 'something',
    },
    fileData: { selected: null, files: []},
};
const mockStore = configureStore(initialState);

test('should render Homepage succesfully', () => { 
    const store = mockStore(initialState);
    const component = shallow(<Provider store={store}><Homepage/></Provider>);
    expect(component.html()).toMatchSnapshot();
});

test('render should contain all div items specified', () => {
    const divs = ['homepageContainer', 'logoContainer', 'logo', 'createButton', 'uploadButton'];
    const store = mockStore(initialState);
    const component = mount(<Provider store={store}><Homepage/></Provider>);

    divs.forEach(divName => {
        const foundBool = component.find(`.${divName}`) || component.find(`#${divName}`)
        expect(foundBool.first().props().className);
    })
});

test('should open CreateModal on createButton click', () => {
    const store = mockStore(initialState);
    const component = mount(<Provider store={store}><Homepage/></Provider>);
    const createButton = component.find('#createButton').first();
    createButton.simulate('click');
    expect(component.html()).toMatchSnapshot();
});

test('should close CreateModal on close button click', () => {
    const store = mockStore(initialState);
    const component = mount(<Provider store={store}><Homepage /></Provider>);
    const createButton = component.find('#createButton').first();
    createButton.simulate('click');
    const closeButton = component.find('.css-1ii3p2c').first();
    closeButton.simulate('click');
    expect(component.html()).toMatchSnapshot();
});

test('should open Upload Modal on Upload Modal click', () => {
    const store = mockStore(initialState);
    const component = mount(<Provider store={store}><Homepage/></Provider>);
    const uploadButton = component.find('#uploadButton').first();
    uploadButton.simulate('click');
    expect(component.html()).toMatchSnapshot();
})

test('should open Upload Modal on Upload Modal click', () => {
    const setAlertNone = jest.fn();
    const store = mockStore(initialState);
    const component = mount(<Provider store={store}><Homepage /></Provider>);
    const uploadButton = component.find('#uploadButton').first();
    uploadButton.simulate('click');
    const closeButton = component.find('.css-1ii3p2c').first();
    closeButton.simulate('click');
    expect(component.html()).toMatchSnapshot();
})