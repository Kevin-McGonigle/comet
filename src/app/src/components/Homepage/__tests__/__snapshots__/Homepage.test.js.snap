// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should close CreateModal on close button click 1`] = `"<div class=\\"homepageContainer\\"><div class=\\"topBar\\"><p>by James Miles &amp; Kevin Mcgonigle</p></div><div class=\\"logoContainer\\"><h1>comet.</h1><h2>code metrics - made simple</h2><div class=\\"logo\\"><img src=\\"comet.png\\" width=\\"200px\\" height=\\"200px\\"><div><button class=\\"css-69cngj createButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"createButton\\">Create</button><button class=\\"css-69cngj uploadButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"uploadButton\\">Upload</button></div></div></div></div>"`;

exports[`should open CreateModal on createButton click 1`] = `"<div class=\\"homepageContainer\\"><div class=\\"topBar\\"><p>by James Miles &amp; Kevin Mcgonigle</p></div><div class=\\"logoContainer\\"><h1>comet.</h1><h2>code metrics - made simple</h2><div class=\\"logo\\"><img src=\\"comet.png\\" width=\\"200px\\" height=\\"200px\\"><div><button class=\\"css-69cngj createButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"createButton\\">Create</button><button class=\\"css-69cngj uploadButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"uploadButton\\">Upload</button></div></div></div></div>"`;

exports[`should open Upload Modal on Upload Modal click 1`] = `"<div class=\\"homepageContainer\\"><div class=\\"topBar\\"><p>by James Miles &amp; Kevin Mcgonigle</p></div><div class=\\"logoContainer\\"><h1>comet.</h1><h2>code metrics - made simple</h2><div class=\\"logo\\"><img src=\\"comet.png\\" width=\\"200px\\" height=\\"200px\\"><div><button class=\\"css-69cngj createButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"createButton\\">Create</button><button class=\\"css-69cngj uploadButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"uploadButton\\">Upload</button></div></div></div></div>"`;

exports[`should open Upload Modal on Upload Modal click 2`] = `"<div class=\\"homepageContainer\\"><div class=\\"topBar\\"><p>by James Miles &amp; Kevin Mcgonigle</p></div><div class=\\"logoContainer\\"><h1>comet.</h1><h2>code metrics - made simple</h2><div class=\\"logo\\"><img src=\\"comet.png\\" width=\\"200px\\" height=\\"200px\\"><div><button class=\\"css-69cngj createButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"createButton\\">Create</button><button class=\\"css-69cngj uploadButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"uploadButton\\">Upload</button></div></div></div></div>"`;

exports[`should render Homepage succesfully 1`] = `"<div class=\\"homepageContainer\\"><div class=\\"topBar\\"><p>by James Miles &amp; Kevin Mcgonigle</p></div><div class=\\"logoContainer\\"><h1>comet.</h1><h2>code metrics - made simple</h2><div class=\\"logo\\"><img src=\\"comet.png\\" width=\\"200px\\" height=\\"200px\\"/><div><button class=\\"css-69cngj createButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"createButton\\">Create</button><button class=\\"css-69cngj uploadButton ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_48px ðŸ“¦ltr-spc_-0-05px ðŸ“¦btrr_4px ðŸ“¦bbrr_4px ðŸ“¦btlr_4px ðŸ“¦bblr_4px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_24px ðŸ“¦pl_24px ðŸ“¦ml_0px ðŸ“¦mr_12px ðŸ“¦mb_0px ðŸ“¦h_48px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" id=\\"uploadButton\\">Upload</button></div></div></div></div>"`;
